app/controllers/setup_controller.rb

```diff
@@ -51,6 +52,11 @@ def welcome
 
     # container runtime setting
     @cri = Pillar.value(pillar: :container_runtime) || "docker"
+
+    # allow adding system certificate: required if a user uses CPI with a
+    # self-signed certificate
+    @system_certificate = SystemCertificate.new
+    @cert = Certificate.new
```

Comment:
Be aware that the user can go back and forth through the pages before the bootstrap itself and any previous value set should be there when the user goes to that specific page.

Reply:
Good point, I updated it to use the `session` to store a entered `form`.

I only tested it with the browser's back, which worked, not with our `back` buttons.